### PR TITLE
Set the item when a Z move is used

### PR DESCRIPTION
**Pull request by urkerab**
Status moves are a bit of a kludge but I don't have any better ideas.

### CODE REVIEW

File: js/battle.js [hunk 1]
```diff
@@ -3374,6 +3374,15 @@ var Battle = (function () {
 					this.message(pokemon.getName() + ' used <strong>' + move.name + '</strong>!');
 				}
 				if (!fromeffect.id || fromeffect.id === 'pursuit') {
+					if (move.name.slice(0, 2) === 'Z-') {
+						move = Tools.getMove(move.name.slice(2));
+						for (item in window.BattleItems) {
```
Comment (ascriptmaster) [inline js/battle.js]: `for (let item in window.BattleItems)` should fix the build error

Comment (Slayer95) [inline js/battle.js]: ``var``!

File: js/battle.js [hunk 2]
```diff
@@ -3374,6 +3374,15 @@ var Battle = (function () {
...
+							if (BattleItems[item].zMoveType === move.type) pokemon.item = item;
+						}
+					}
+					if (move.isZ) {
```
Comment (Zarel) [inline js/battle.js]: This should probably be first

File: js/battle.js [hunk 3]
```diff
@@ -3374,6 +3374,15 @@ var Battle = (function () {
 					this.message(pokemon.getName() + ' used <strong>' + move.name + '</strong>!');
 				}
 				if (!fromeffect.id || fromeffect.id === 'pursuit') {
+					if (move.name.slice(0, 2) === 'Z-') {
```
Comment (Zarel) [inline js/battle.js]: And then this would be an `else if`

File: js/battle.js (hunk 2 above)
Comment (urkerab) [inline js/battle.js]: (An earlier iteration set `isZ` to the appropriate item for the status move.)

**Comment by urkerab** [comment]
> Status moves are a bit of a kludge but I don't have any better ideas.

`kwargs.anim` has the true move in that case; would that be less of a kludge?

**Pull request merged**